src/java.base/share/classes/sun/net/www/protocol/jar/Handler.java

```diff
@@ -193,19 +198,18 @@ private String parseContextSpec(URL url, String spec) {
         if (spec.startsWith("/")) {
             int bangSlash = indexOfBangSlash(ctxFile);
             if (bangSlash == -1) {
-                throw new NullPointerException("malformed " +
-                                               "context url:" +
-                                               url +
-                                               ": no !/");
+                throw new NullPointerException(
+                    formatMsg("malformed context url%s : no !/",
+                              filterJarName(url.toString()).prefixWith(": ")));
```

Comment:
It's not clear whether `url` could be `null` here, so to sidestep the question maybe use `String::valueOf` rather than `Object::toString`.